crates/accelerate/src/synthesis/cnot_synthesis.rs

```diff
@@ -0,0 +1,163 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2024
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use hashbrown::HashMap;
+use ndarray::{s, Array1, Array2, Axis};
+use numpy::PyReadonlyArray2;
+use smallvec::smallvec;
+use std::cmp;
+
+use qiskit_circuit::circuit_data::CircuitData;
+use qiskit_circuit::operations::{Param, StandardGate};
+use qiskit_circuit::Qubit;
+
+use pyo3::prelude::*;
+
+/// Mutate a matrix inplace by adding the value of the ``ctrl`` row to the
+/// ``target`` row. If ``add_columns`` is true, add columns instead of rows.
+fn _add(mat: &mut Array2<bool>, add_columns: &bool, ctrl: usize, trgt: usize) {
```

Comment:
is it the same function as `_row_op` and `_col_op` in #12456 ?
do we want to keep just one of these suggestions?

Reply:
in this case we should discuss the preferred API (let's talk offline)